### PR TITLE
cb110 Adjust the DB mock for trace

### PR DESCRIPTION
**Pull request by Brandonz07**
The original version of the “chat” component in the “management” field uses a message mock to render and save the messages, however, it should render and save the messages based on the “trace” mock. Update the behavior of the chat component to do that.

This task includes the deletion of the message mock and the adjustments of the trace entity.

image

### CODE REVIEW

File: src/pages/board/outlets/financialReporting/management/index.tsx [hunk 1]
```diff
@@ -33,38 +43,48 @@ export const Management = () => {
...
-
-      setMessages(message);
-    });
-  }, []);
-
-  useEffect(() => {
-    localforage.setItem("messages", messages).catch((err) => {
-      console.error("Error al guardar mensajes:", err);
+      if (Array.isArray(data)) {
+        const trace = data.find(
```
Comment (Andresbl123) [inline src/pages/board/outlets/financialReporting/management/index.tsx:47]: esta parte no me queda muy clara, no entiendo por que solo se traer el primer elemento que coincida con el id, aquí se pierde el resto de la data asociada al cliente

File: src/pages/board/outlets/financialReporting/management/index.tsx [hunk 2]
```diff
@@ -33,38 +43,48 @@ export const Management = () => {
...
-  }, [messages]);
+  }, [id]);
 
   const handleFormSubmit = (e: React.FormEvent<HTMLFormElement>) => {
     e.preventDefault();
     sendMessage();
   };
 
-  const sendMessage = () => {
+  const sendMessage = async () => {
```
Comment (Andresbl123) [inline src/pages/board/outlets/financialReporting/management/index.tsx:61]: yo creo que de acuerdo ha lo hablado en la reunión esta función no es necesaria

File: src/pages/board/outlets/financialReporting/management/index.tsx [hunk 3]
```diff
@@ -21,9 +20,20 @@ interface MessageType {
   text: string;
 }
 
-interface MessageTypedos {
+interface TraceType {
```
Comment (Andresbl123) [inline src/pages/board/outlets/financialReporting/management/index.tsx:23]: esta interface, creo que tampoco va, y de ser necesaria una interface para `trece`, crearla en directorio de `services`

**Review by Andresbl123 — APPROVED** [review]

File: src/mocks/financialReporting/trace.mock.ts [hunk 4]
```diff
@@ -4,103 +4,101 @@ novelty, executed_task,
...
-      execution_date: "23/dic/2023",
-      justification: "Final approval given",
-      decision_taken_by_user: "approved",
-      trace_type: "document",
-      read_novelty: "N",
-    },
-  ],
-};
+export const traceMock = [
+  {
```
Comment (carin91) [inline src/mocks/financialReporting/trace.mock.ts:8]: Adjust the build Data util to include an additional prop that defines whether or not the build data will include an id or not.
image

In the case of the trace mock, the id should not be included.

**Review by carin91 — CHANGES_REQUESTED** [review]

File: src/mocks/financialReporting/trace.mock.ts (hunk 4 above)
Comment (carin91) [reply src/mocks/financialReporting/trace.mock.ts:8]: The change is pending:
image

Comment (Brandonz07) [reply src/mocks/financialReporting/trace.mock.ts:8]: The trace is working correctly, try logging out

image

**Review by carin91 — APPROVED** [review]

**Pull request merged**